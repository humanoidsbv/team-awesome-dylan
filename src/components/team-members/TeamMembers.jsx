import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

import styles from './TeamMembers.module.css';
import TeamMember from '../team-member/TeamMember';
import TeamMemberForm from '../team-member-form/TeamMemberForm';

function TeamMembers({ createTeamMember, fetchTeamMembers, teamMembers }) {
  useEffect(() => {
    fetchTeamMembers();
  }, []);

  return (
    <div>
      <TeamMemberForm
        createTeamMember={newTeamMember => createTeamMember(newTeamMember)}
      />
      <div className={styles.headerTeamMembers}>
        <span className={styles.headerTeamMembersText}> All Humanoids </span>
        <button type="button" className={styles.newHumanoidButtonDesktop}>
          + New Humanoid
        </button>
        <select type="button" className={styles.sortTeamMembersSelector}>
          <option value="">Sort by:</option>
        </select>
      </div>
      <div className={styles.teamMembersContainer}>
        <button type="button" className={styles.newHumanoidButton}>
          + New Humanoid
        </button>
        {teamMembers.map(
          ({
            currentEmployer,
            employeeFunction,
            employeeNumber,
            firstName,
            lastName,
            startingDate
          }) => (
            <TeamMember
              currentEmployer={currentEmployer}
              employeeFunction={employeeFunction}
              employeeNumber={employeeNumber}
              firstName={firstName}
              lastName={lastName}
              startingDate={startingDate}
            />
          )
        )}
      </div>
    </div>
  );
}

TeamMembers.propTypes = {
  teamMembers: PropTypes.arrayOf(
    PropTypes.shape({
      firstName: PropTypes.string
    })
  ),
  createTeamMember: PropTypes.func.isRequired,
  fetchTeamMembers: PropTypes.func.isRequired
};

TeamMembers.defaultProps = {
  teamMembers: []
};

export default TeamMembers;
