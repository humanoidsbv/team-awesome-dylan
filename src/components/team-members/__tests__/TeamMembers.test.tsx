import * as React from 'react';
import { shallow } from 'enzyme';

import TeamMembers from '../TeamMembers';

const mockedTeamMembers = [{
  address: 'string',
  currentClient: 'string',
  emailAddress: 'string',
  employeeFunction: 'string',
  employeeNumber: 'string',
  firstName: 'string',
  id: 3,
  key: 4,
  lastName: 'string',
  locality: 'string',
  postalCode: 'string',
  startingDate: 'string'
}];

it('matches snapshot', () => {
  const teamMembers = shallow(
    <TeamMembers
      createTeamMember={({}) => null}
      fetchTeamMembers={()=> null}
      sortDirection={true}
      sortTeamMembersByField={(string: string) => null }
      sortTeamMembersDirection={() => ({ type: 'bla' }) }
      teamMembers ={mockedTeamMembers}
    />
  );
  expect(teamMembers).toMatchSnapshot();
});
