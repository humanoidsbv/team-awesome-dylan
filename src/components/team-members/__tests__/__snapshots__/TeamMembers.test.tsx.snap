// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <TeamMemberForm
    createTeamMember={[Function]}
    isTeamMemberFormVisible={false}
    toggleTeamMemberForm={[Function]}
  />
  <div
    className="headerTeamMembers"
  >
    <span
      className="headerTeamMembersText"
    >
       All Humanoids 
    </span>
    <button
      className="newHumanoidButtonHeader "
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      + New Humanoid
    </button>
    <div
      className="sortingWrapper"
    >
      <select
        className="sortTeamMembersSelector"
        id="sortSelect"
        onChange={[Function]}
      >
        <option
          value="firstName"
        >
          Sort by: First name
        </option>
        <option
          value="lastName"
        >
          Sort by: Last name
        </option>
        <option
          value="employeeFunction"
        >
          Sort by: Employee function
        </option>
        <option
          value="employeeNumber"
        >
          Sort by: Employee number
        </option>
        <option
          value="currentClient"
        >
          Sort by: Current client
        </option>
        <option
          value="startingDate"
        >
          Sort by: Starting date
        </option>
      </select>
      <button
        className="sortDirectionToggle"
        onClick={[Function]}
        type="button"
      >
        <test-file-stub
          className="sortDirectionAscending"
        />
      </button>
    </div>
  </div>
  <div
    className="teamMembersContainer"
  >
    <TeamMember
      currentClient="string"
      employeeFunction="string"
      employeeNumber="string"
      firstName="string"
      key="3"
      lastName="string"
      startingDate="string"
    />
  </div>
</div>
`;
